ssh_automated_tests/core/connection.py

```diff
@@ -10,13 +10,17 @@ class Connection(ABC):
     def connect(self):
         pass
 
+    @abstractmethod
+    def disconnect(self):
+        pass
 
-class SshConnection:
+
+class SshConnection(Connection):
```

Comment:
Looks like `open_shell` and `open_sftp` has to be separate implementations of a `Connection`.